src/Box/Box.ts

```diff
@@ -48,4 +48,8 @@ const Box: PolyComponent<'div', Props> = styled(TextDivWithColorPropFix)<Props>`
   ${width};
 `;
 
-export default Box;
+const PolyBox: PolyComponent<'div', Props> = Box;
+
+export default Object.assign(PolyBox, {
+  withComponent: Box.withComponent
```

Comment:
I had to rework the internals a bit here in order to expose `withComponent` since `DescriptionList` and friends build on `Box` in that way.